Automattic/jetpack
# Add modules store tests

Fixes https://github.com/Automattic/jetpack/issues/34589

## Proposed changes:

Removed some unnecessary logic from updateJetpackModuleStatus. We will always try to set new data.
Main focus of PR is covering this `store` with tests

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Checkout this branch locally and run the tests `jetpack test js-packages/shared-extension-utils`

If decided to test general behavior next steps could be used:
- Spin up your site with local changes
- Navigate to `wp-admin/admin.php?page=jetpack_modules` and make sure Contact Form module is disabled
- Due to https://github.com/Automattic/jetpack/issues/34793, need to build all modules that use this package for proper testing (jetpack build packages/forms, jetpack build plugins/jetpack, jetpack build packages/videopress)
- Create a new post and add contact form. Activate module. Observe the behavior.
- Verify self-hosted and Simple/Atomic work normally.
